### PR TITLE
Shorten output of E0391

### PR DESCRIPTION
**Pull request by estebank**
Use the shorter `def_span` on the impl-Trait cyclic reference errors.

### CODE REVIEW

File: src/test/ui/impl-trait/auto-trait-leak.stderr [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  --> $DIR/auto-trait-leak.rs:52:1
+   |
+52 | fn cycle1() -> impl Clone {
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^ cyclic reference
+   |
+note: the cycle begins when processing `cycle1`...
+  --> $DIR/auto-trait-leak.rs:52:1
+   |
+52 | fn cycle1() -> impl Clone {
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (kennytm) [inline src/test/ui/impl-trait/auto-trait-leak.stderr]: Actual output differs from expected output. Please update.

```diff
 error[E0391]: unsupported cyclic reference between types/traits detected
   --> $DIR/auto-trait-leak.rs:52:1
    |
-52 | fn cycle1() -> impl Clone {
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^ cyclic reference
+52 |   fn cycle1() -> impl Clone {
+   |  _-^^^^^^^^^^^^^^^^^^^^^^^^
+53 | |     //~^ ERROR unsupported cyclic reference between types/traits detected
+54 | |     //~| cyclic reference
+55 | |     //~| NOTE the cycle begins when processing `cycle1`...
+...  |
+60 | |     Rc::new(Cell::new(5))
+61 | | }
+   | |_- cyclic reference
```

**Review by kennytm — COMMENTED** [review]
The output of `issue-23305.rs` is also affected.

```diff
 error[E0391]: unsupported cyclic reference between types/traits detected
   --> $DIR/issue-23305.rs:15:12
    |
 15 | impl ToNbt<Self> {}
    |            ^^^^ cyclic reference
    |
 note: the cycle begins when processing `<impl at $DIR/issue-23305.rs:15:1: 15:20>`...
   --> $DIR/issue-23305.rs:15:1
    |
 15 | impl ToNbt<Self> {}
-   | ^^^^^^^^^^^^^^^^^^^
+   | ^^^^^^^^^^^^^^^^
    = note: ...which then again requires processing `<impl at $DIR/issue-23305.rs:15:1: 15:20>`, completing the cycle.
 
 error: aborting due to previous error
```

File: src/test/ui/impl-trait/auto-trait-leak.stderr (hunk 1 above)
Comment (estebank) [reply src/test/ui/impl-trait/auto-trait-leak.stderr]: Missed the label. Done.

**Pull request merged**